File: DataSets/DataSet-Old/LeetCodeTasksDevelop/LeetCodeTasks/3005_count-elements-with-maximum-frequency/solutions/cpp/solution.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    int maxFrequencyElements(std::vector<int>& nums) {
        std::unordered_map<int, int> freq;
        int maxFreq = 0;
        for (int num : nums) {
            maxFreq = std::max(maxFreq, ++freq[num]);
        }
        int totalCount = 0;
        for (auto& pair : freq) {
            if (pair.second == maxFreq) {
                totalCount += maxFreq;
            }
        }
        return totalCount;
    }
};

File: DataSets/DataSet-Old/LeetCodeTasksDevelop/LeetCodeTasks/3005_count-elements-with-maximum-frequency/solutions/java/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int maxFrequencyElements(int[] nums) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();
        for (int num : nums) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        int maxFrequency = 0;
        for (int freq : frequencyMap.values()) {
            if (freq > maxFrequency) {
                maxFrequency = freq;
            }
        }

        int count = 0;
        for (Map.Entry<Integer, Integer> entry : frequencyMap.entrySet()) {
            if (entry.getValue() == maxFrequency) {
                count += entry.getValue();
            }
        }

        return count;
    }
}
